test/cmdlineTests/~bytecode_equivalence_with_unused_contracts/test.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from pathlib import Path
+from textwrap import dedent
+
+# pylint: disable=wrong-import-position
+PROJECT_ROOT = Path(__file__).parents[3]
+sys.path.insert(0, str(PROJECT_ROOT / 'scripts'))
+
+from common.cmdline_helpers import add_preamble
+from common.cmdline_helpers import inside_temporary_dir
+from common.cmdline_helpers import save_bytecode
+from common.cmdline_helpers import solc_bin_report
+from common.git_helpers import git_diff
+from splitSources import split_sources
+
+
+@inside_temporary_dir(Path(__file__).parent.name)
+def test_bytecode_equivalence():
+    source_file_path = Path(__file__).parent / 'inputs.sol'
+    split_sources(source_file_path, suppress_output=True)
+    add_preamble(Path.cwd())
+
+    solc_binary = os.environ.get('SOLC')
+    if solc_binary is None:
+        raise RuntimeError(dedent("""\
+            `solc` compiler not found.
+            Please ensure you set the SOLC environment variable
+            with the correct path to the compiler's binary.
+        """))
+
+    # Repro for https://github.com/ethereum/solidity/issues/14829
+    save_bytecode(Path('A.bin'),
+                  solc_bin_report(solc_binary,
+                                  [Path('A.sol')],
```

Comment:
Souldn't this be `B` instead?
```suggestion
                                  [Path('B.sol')],
```

Reply:
Ha, yes, you're right; but unfortunately, doesn't seem to solve my problem :/

Reply:
Oh, but what is the problem then? If you want, we can discuss that in the Matrix dev channel, just ping me there.

Reply:
I think that the problem was that this does not actually reproduce the problem on `develop`. And I just posted a solution to that on the channel:

> Either remove `add_preamble()` or reduce the number of dummy contracts. Removing two last ones does it for me.

This shorter repro requires a specific shift in AST IDs so even adding a pragma apparently breaks it. Maybe it would actually better to use the more complex one. It seemed more robust to me.

Reply:
By the way, warnings can be suppressed by adding something like `a_1; a_2; a_3;` at the end of the function and making it `pure`. Which is not great, but still better than having the warnings :)